File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int ans = 0;
        unordered_set<int> set;
        for(auto num: nums){
            set.insert(num);
        }
        for(auto num: nums){
            if(set.find(num-1) == set.end()){
                int curr_len = 0;
                int curr_num = num;
                while(set.find(curr_num) != set.end()){
                    curr_num++;
                    curr_len++;
                }
                ans = max(ans, curr_len);
            }
        }
        return ans;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        st = set(nums)
        ans = 0

        for num in st:
            if num-1 not in st:
                curr_len = 0
                x = num
                while x in st:
                    x += 1
                    curr_len += 1

                ans = max(ans, curr_len)

        return ans
        